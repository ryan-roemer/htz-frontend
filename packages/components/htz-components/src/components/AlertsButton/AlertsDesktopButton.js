import React from 'react';
import { createComponent, } from 'react-fela';
import AlertsButton, { authorPropTypes, } from './_AlertsButton';
import IconMail from '../Icon/icons/IconMail';
import Color from '../Color/Color';

const AlertsDesktopButtonStyled = createComponent(
  ({ theme, }) => ({
    display: 'block',
    width: '100%',
    textAlign: 'end',
  }),
  AlertsButton,
  [ 'author', ]
);

function AlertsDesktopButton({ author, }) {
  return (
    <AlertsDesktopButtonStyled author={author}>
      <Color
        color="tertiary"
        miscStyles={{
          fontWeight: 700,
          type: -1,
        }}
      >
        <IconMail size={3} miscStyles={{ marginEnd: '1rem', }} />
        התראות במייל
      </Color>
    </AlertsDesktopButtonStyled>
  );
}

AlertsDesktopButton.propTypes = authorPropTypes;

export default AlertsDesktopButton;
