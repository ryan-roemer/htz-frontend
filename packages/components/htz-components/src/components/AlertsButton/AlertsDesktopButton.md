<!-- START doctoc generated TOC please keep comment here to allow auto update -->
<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->

**Table of Contents**

- [AlertsDesktopButton Component](#alertsdesktopbutton-component)

<!-- END doctoc generated TOC please keep comment here to allow auto update -->

### AlertsDesktopButton Component

A button that opens a registration form for author-alerts.(Email newsletters)

```jsx
<div dir="rtl">
  <AlertsDesktopButton
    author={{
      name: 'Avi Kaufman',
    }}
  />
</div>
```
