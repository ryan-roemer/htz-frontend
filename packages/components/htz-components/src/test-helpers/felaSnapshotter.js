import React from 'react';
import createRenderer from '../components/StyleProvider/createRenderer';
import StyleProvider from '../components/StyleProvider/StyleProvider';
import createFelaSnapshotter from './createFelaSnapshotter';

const testRenderer = createRenderer({ isRtl: true, });
const felaSnapshotter = createFelaSnapshotter(StyleProvider, testRenderer);

export default felaSnapshotter;
