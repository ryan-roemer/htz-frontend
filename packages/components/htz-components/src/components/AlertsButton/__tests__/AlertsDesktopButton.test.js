import React from 'react';
import felaSnapshotter from '../../../test-helpers/felaSnapshotter';
import AlertsDesktopButton from '../AlertsDesktopButton'; // eslint-disable-line import/no-named-as-default

describe('AlertsDesktopButton component', () => {
  it('renders correctly', () => {
    const author = {
      name: 'Avi Kaufman',
    };
    
    const snapshot = felaSnapshotter(
      <AlertsDesktopButton author={author}>This is a subtitle</AlertsDesktopButton>
    );
    expect(snapshot).toMatchSnapshot();
  });
});
