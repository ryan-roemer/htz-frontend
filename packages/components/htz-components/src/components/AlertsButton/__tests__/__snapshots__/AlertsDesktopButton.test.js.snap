// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlertsDesktopButton component renders correctly 1`] = `
Object {
  "component": <div>
    <button
      aria-describedby="alerts_btn_description"
      aria-expanded={false}
      className="htz-a htz-b htz-c"
      onClick={[Function]}
    >
      <span
        className="htz-d htz-e htz-f htz-g htz-h htz-i htz-j htz-k htz-l htz-m htz-n"
        id="alerts_btn_description"
      >
        קבל התראות בתיבת הדואר האלקטרוני שלך עבור כתבות מ 
        Avi Kaufman
      </span>
      <span
        className="htz-o htz-p htz-q htz-r htz-s htz-t"
      >
        <svg
          className="htz-u htz-v htz-w htz-x htz-y"
          height="1em"
          viewBox="0 0 256 256"
          width="1em"
        >
           
          <path
            d="M242.4 49l-114.9 90.3L13.8 49h228.6zm-71.3 71.4l72.2 68.3-8.2 8.7-73.5-69.5-34 26.8-34.1-27.1-73.1 69.6-8.3-8.7L83.9 120 8 59.6V207h240V59.9l-76.9 60.5z"
            fill="currentColor"
          />
        </svg>
        התראות במייל
      </span>
    </button>
  </div>,
  "styles": ".htz-a{display:block}.htz-b{width:100%}.htz-c{text-align:left}.htz-d{border:0px}.htz-e{clip:rect(0 0 0 0)}.htz-f{-webkit-clip-path:inset(50%)}.htz-g{clip-path:inset(50%)}.htz-h{height:1px}.htz-i{margin:-1px}.htz-j{overflow:hidden}.htz-k{padding:0rem}.htz-l{position:absolute}.htz-m{width:1px}.htz-n{white-space:nowrap}.htz-o{color:#A8001C}.htz-p{font-weight:700}.htz-u{vertical-align:-0.15em}.htz-v{color:inherit}.htz-w{fill:#FFF}.htz-x{font-size:3rem}.htz-y{margin-left:1rem}@media (max-width: 79.9375em){.htz-q{font-size:2.3333333333333335rem}.htz-r{line-height:1.2857142857142858em}}@media (min-width: 80em){.htz-s{font-size:2.2857142857142856rem}.htz-t{line-height:1.3125em}}",
}
`;
